### DISCUSSION TITLE
Is there a 2D drawing widget?

### DISCUSSION THREAD

**Discussion by bluthej** [Kitware/trame#507] [Q&A] [answered]
Hi !

I was wondering if there is a widget that would allow the user to draw 2D shapes like polygons using the mouse?
I could not find something like this in the docs but maybe I just did not look in the right places.

**Comment by jourdain** [comment, marked as answer]
Currently I'm not aware of any widget like that. But this could be created or even better, we could integrate something that already exist from JS or Vue.js.

Do you know any library that does what you are looking for?

**Reply by bluthej** [reply]
Well at my company we've been playing around with [react-konva](https://www.npmjs.com/package/react-konva), it does the job for us, do you think it could be integrated with trame?

**Reply by jourdain** [reply]
It looks like plain svg could work for you. Just use vue reactivity to drive and control the various elements.

**Reply by Jimmy-KL** [reply]
We use [PixiJS](https://pixijs.com/) for 2D drawing.
